code/modules/mob/living/carbon/alien/diona/diona_nymph.dm

```diff
@@ -191,22 +191,14 @@
 	vessel = new/datum/reagents(600)
 	vessel.my_atom = src
 
-	vessel.add_reagent(/datum/reagent/blood, 560)
+	vessel.add_reagent(/decl/reagent/blood, 560, temperature = species?.body_temperature)
 	fixblood()
 
 /mob/living/carbon/alien/diona/proc/fixblood()
-	for(var/datum/reagent/blood/B in vessel.reagent_list)
-		if(B.type == /datum/reagent/blood)
-			B.data = list(
-				"donor" = WEAKREF(src),
-				"species" = species.name,
-				"blood_DNA" = name,
-				"blood_colour" = species.blood_color,
-				"blood_type" = null,
-				"resistances" = null,
-				"trace_chem" = null
-			)
-			B.color = B.data["blood_colour"]
+	if(!REAGENT_DATA(vessel, /decl/reagent/blood))
+		return
+	var/list/new_blood_data = get_blood_data()
+	vessel.reagent_data[/decl/reagent/blood] = vessel.reagent_data[/decl/reagent/blood] ^ new_blood_data | new_blood_data
```

Comment:
Isn't the purpose of this proc to reset the blood to a known good state? Why are we keeping parts of the original blood?

It looks like the original version kept nothing.

Reply:
It resets the blood-specific data, which would also reset thermal energy. This meant blood was supercooled.

Reply:
Makes sense.